src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -548,16 +548,21 @@ def list_resource_links(scope=None, filter_string=None):
         return links_client.list_at_source_scope(scope, filter=filter_string)
     return links_client.list_at_subscription(filter=filter_string)
 
+def _validate_resource_inputs(resource_group_name, resource_provider_namespace,
+                              resource_type, resource_name):
+    if resource_group_name is None:
+        raise CLIError('--resource-group/-g is required.')
+    if resource_type is None:
+        raise CLIError('--resource-type is required')
+    if resource_name is None:
+        raise CLIError('--name/-n is required')
+    if resource_provider_namespace is None:
+        raise CLIError('--namespace is required')
```

Comment:
If you provided just a name, you would have to run this three times to know the three parameters you were missing. I would recommend something like:
```Python
missing = []
if not resource_group_name: missing.append('--resource-group/-n')
if not resource_type: missing.append('--resource-type')
...
if missing:
  raise CLIError('Missing the following: {}. Alternatively, use --id'.format(' '.join(missing)))
```

Alternatively, simply reuse the usage statement from the old code, as that makes clear what is required as well as the alternative.